### PR TITLE
GHA: Run tests on multiple platforms

### PR DESCRIPTION
**Pull request by Robbepop**
Carved out from #274.

### CODE REVIEW

**Comment by Robbepop** [comment]
GHA CI passes fully. Note that some checkmarks are not there since our branch protection settings need to be updated in order to resemble the new CI passes.

**Comment by athei** [comment]
Can you add `--test-threads 1` here? Adding this can also fail tests if they make use of thread local storage (some do).

**Comment by Robbepop** [comment]
> Can you add `--test-threads 1` here? Adding this can also fail tests if they make use of thread local storage (some do).

You mean as an entire new test pass or add to existing test pipeline?

**Comment by athei** [comment]
An extra pipeline which runs `windows + virtual_mem` with `--test-threads 1`.

**Comment by Robbepop** [comment]
> An extra pipeline which runs `windows + virtual_mem` with `--test-threads 1`.

The `windows.latest` + `features virtual_memory` job now runs using `--test-threads 1` whereas everything else runs in parallel.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -58,9 +67,10 @@ jobs:
       - uses: actions-rs/cargo@v1
         env:
           RUSTFLAGS: '--cfg debug_assertions'
+          RUSTC_FLAGS: ${{ matrix.rustc-args }}
```
Comment (athei) [inline .github/workflows/rust.yml:70]: The name is wrong. Those flags aren't passed to rustc. They are passed to the test binary.

Comment (Robbepop) [reply .github/workflows/rust.yml:70]: is `TEST_FLAGS` better?

Comment (athei) [reply .github/workflows/rust.yml]: Yes! But you also need to rename `rust-args` to `test-args`. :D

**Review by athei — APPROVED** [review]

**Pull request merged**